oauth2_provider/models.py

```diff
@@ -434,8 +435,15 @@ def get_refresh_token_model():
     return apps.get_model(oauth2_settings.REFRESH_TOKEN_MODEL)
 
 
-def clear_expired():
-    now = timezone.now()
+def clear_expired(before=None):
+    """Clear expired tokens, add custom date to limit clear tokens before this date."""
+    if not before:
```

Comment:
I feel this whole block could be simplified to:
```suggestion
    if not tokens_expired_before_date:
        tokens_expired_before_date = timezone.now()
    if isinstance(tokens_expired_before_date, date) is False:
            logger.info('Not valid datetime provided.')
            return  
```